contents/docs/how-posthog-works/ingestion-pipeline.mdx

```diff
@@ -0,0 +1,172 @@
+| ------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------ |
+| 1. Neither `$anon_distinct_id` nor `$distinct_id` have been associated with a Person       | Create a new Person and add a mapping in PostgreSQL to associate this `$distinct_id` with the new `person_id`      |
+| 2. _Only one_ of `$anon_distinct_id` and `$distinct_id` have been associated with a Person | Create a new mapping to associate the `$distinct_id` and `$anon_distinct_id` with the already existing `person_id` |
+| 3. _Both_ `$anon_distinct_id` _and_ `$distinct_id` have been associated with a Person      | We will [merge these two people](#merging-two-persons) and associate all future events with the first Person       |
+
+> **Note**: In the case of an event sent from the backend, there will be no `$anon_distinct_id` field present. Instead, the same lookups and checks will be performed, but only on the `$distinct_id` which is required for all backend events
+
+###### Merging two Persons
+
+In the third scenario, where we have inadvertently created two Persons for the same user, we will need to merge them. In the case of an `$identify` call, we will create a mapping from the `distinct_id` of the anonymous user to the non-anonymous user.
+If there are any conflicts when merging Person properties for these two users, the values from the non-anonymous user will take precedence.
+We choose to prioritize the history of the non-anonymous user, as it is far more likely that this user will have a history of previous events associated with them that we want to preserve.
+
+###### Consequences of merging
+
+This scenario typically isn't ideal, as merging two users will only affect events that are sent in the future.
+As a result, we will be left with two separate Persons that should be associated with the same user, but instead will be treated as entirely unique users.
+The only way to fully 'merge' these two people would be to go back and rewrite the `person_id` field on past events, which is not practical or performant.
+
+For more information on how these merges can affect the results of your queries, take a look at this page on how [querying data](/data/how-posthog-works/queries) works.
+
+###### Avoiding a merge
+
+Avoiding this scenario is the primary goal of the event buffer: by [buffering certain events](#1-event-buffer), we can avoid creating a duplicate Person when other events arrive before an `$identify` call.
+
+On the integration side, the best way to avoid this scenario is to send `$identify` events as early into a user's session as possible.
+This really only presents an issue when a user is logging in for the first time from a new device, and does not present any problems with users logging out and logging back in on an already seen device and platform.
```

Comment:
nope - it's a problem always. In fact we don't have a solution for second/third logins (they will be merged and the events that happened before logging in will be as if a different person). But we know that the initial signup is something PostHog users care about a lot more compared to future logins, so we're taking the extra complexity here to make sure we keep the initial signup flow events tied to the same person_id as much as possible.